### ISSUE TITLE
Ignore pattern matching

### ISSUE THREAD

**Issue by schnika** [tolk/tolk#158]
With the recent commit 0a17bbe6c019cb71f14c9f2fe787c2baa8893abc the pattern matching changed in a way that the current master would introduce a breaking change.

I take responsibility for that and would like to have an open discussion about the issues I see with that.

Before this commit (0a17bbe6c019cb71f14c9f2fe787c2baa8893abc) the regex filtered out too many keys.

For example **anything** that contained the ignored key (from position 2 on in the ignore_keys array - e.g. when `config.ignored_keys = ['faker', 'date']`) would actually ignore even partially matching keys. In this example 'birthdate' would get ignored.

With the recent changes made by the mentioned commit, the whole **flattened key** needs to match. This results in nested keys not being ignored anymore (which is in contrast to the documentation as well).

So take this `yml` for example:

```
en:
  faker:
    address: will not be ignored
  foo: ignored
```

`['foo', 'faker']` would actually only filter out these keys :

`foo`

but not `faker.address`.

If the `yml` would look like this:

```
en:
  faker: ignored
  foo: ignored
```

both keys would get ignored. 

In short. Deeper levelled keys are not being ignored.

If we would change the regexp to `\A(foo|bar)` deeper nested keys would be ignored BUT also keys like `fakergod` or `foobar`.

As you can see the solution is a bit more complex (correct me if I am wrong).

I would like to kickoff a new discussion with a reference to this rubular https://rubular.com/r/k0UKywfOr3mMew where the different cases are implemented. I am 100% sure that there is a better way of implementing this regexp :).

But maybe the solution is to allow an array of regexps in the ignored keys config and support more advanced configuration through that.

I am open for suggestions and will be happy to contribute to the next iteration.

**Comment by thomasdarde** [comment]
Hello ! 

Thanks for the discussion,
I think that your new regex is definitly a first very good improvment and will gladly merge it and release a 4.0 to signal the breaking change (if you want to write this in the readme or changelog it's perfect)

Having the possibility to pass a regex in the ignored keys arrays would be another improvement in fact (but this can be done on a separate 4.1 release)
